### ISSUE TITLE
Server-side authorize function does not work as advertised in README example

### ISSUE THREAD

**Issue by yoruvo** [Thream/socketio-jwt#21]
## Steps To Reproduce

Using `jwks-rsa` version 1.12.2.

1. Step 1
Try to use the server-side example code from the README.

## The current behavior

![image](https://user-images.githubusercontent.com/48255300/106067124-4c1f2000-60fe-11eb-8c8b-a7abbb4c43d9.png)
![image](https://user-images.githubusercontent.com/48255300/106067134-52ad9780-60fe-11eb-969c-dd9c31568011.png)

Also, if debugging `key`, it returns children `kid`, `alg`, `nbf`, `publicKey` and `getPublicKey`. No "rsaPublicKey".

`decodedToken` is weakly typed.

I managed to get the example to compile with the following code:

```ts
      secret: async (decodedToken) => {
        console.debug("I'm trying to authorize...", decodedToken)
        if (
          decodedToken &&
          typeof decodedToken === "object" &&
          "header" in decodedToken &&
          decodedToken.header.kid
        ) {
          const key = await client.getSigningKeyAsync(decodedToken.header.kid)
          return key.getPublicKey()
        }
        return ""
      },
```

However, the problem that remains is that `io.on("connection"` never fires and there is no sign that the connection is established at all.

The token and authentication in place is the same that already successfully works with my `express-jwt` middleware.

## The expected behavior

The above does not happen.

**Comment by yoruvo** [comment]
I was able to fix it for my use case. The above code snippet removes the TS errors, and the problem with my key was algorithm `RS256` instead of the default. Now, it seems to work.

However there was basically no feedback from the library on what was going on, and whether there was or wasn't a connection. It would be nice to have an option to see why the authorization failed.

**Comment by Divlo** [comment]
Thanks for your report! @yoruvo 
So as I understood you find out how to fix your problem but you would like to have more/better error message so you could fix it more easily.

Sounds good to me, we're open to PR, if you have some time to open a PR to improve that, we would really appreciate it! 👌

**Comment by yoruvo** [comment]
Hi @Divlo,

part of my issue came from my own mistake of using the wrong algorithm, sure. Still, the fact remains that your TS server example in the README doesn't work at all, due to the reasons outlined in the ticket.

Above I wrote some example code that *does* work, but it's quite ugly because your typing for `decodedToken` is either null, string, or a Record<string,any>. Is that supposed to be the case? Is it possible for `decodedToken` to be all these things in a JWT token context? I'm not familiar with the architecture, but it seems too lightly typed.

**Comment by Divlo** [comment]
Indeed, the typings were wrong for `decodedToken`, I released `@thream/socketio-jwt@1.1.1` so now the example in the `README` work as expected, feel free to open others issues if any.
Thanks again for your contribution and hopefully it fix your problem. :tada: @yoruvo
